src/frontend/src/lib/services/request-poh-credential.services.ts

```diff
@@ -0,0 +1,82 @@
+import {
+	INTERNET_IDENTITY_ORIGIN,
+	POH_ISSUER_CANISTER_ID,
+	POH_ISSUER_ORIGIN
+} from '$lib/constants/app.constants';
+import { i18n } from '$lib/stores/i18n.store';
+import { userProfileStore, type UserProfile } from '$lib/stores/settings.store';
+import { toastsError } from '$lib/stores/toasts.store';
+import { Principal } from '@dfinity/principal';
+import { isNullish, nonNullish } from '@dfinity/utils';
+import {
+	requestVerifiablePresentation,
+	type VerifiablePresentationResponse
+} from '@dfinity/verifiable-credentials/request-verifiable-presentation';
+import { get } from 'svelte/store';
+
+// This credential type is defined by the issuer Decide AI
+const POH_CREDENTIAL_TYPE = 'ProofOfUniqueness';
+
+export const requestPohCredential = async ({
+	credentialSubject
+}: {
+	credentialSubject: Principal;
+}): Promise<boolean> => {
+	const i18Copy = get(i18n);
+	return new Promise((resolve, reject) => {
+		const issuerCanisterId = nonNullish(POH_ISSUER_CANISTER_ID)
+			? Principal.fromText(POH_ISSUER_CANISTER_ID)
+			: undefined;
+		if (isNullish(POH_ISSUER_ORIGIN) || isNullish(issuerCanisterId)) {
+			toastsError({
+				msg: { text: i18Copy.auth.error.missing_poh_issuer_origin }
+			});
+			resolve(false);
+			return;
+		}
+		requestVerifiablePresentation({
+			issuerData: {
+				origin: POH_ISSUER_ORIGIN,
+				canisterId: issuerCanisterId
+			},
+			identityProvider: new URL(INTERNET_IDENTITY_ORIGIN),
+			credentialData: {
+				credentialSpec: {
+					credentialType: POH_CREDENTIAL_TYPE,
+					arguments: {}
+				},
+				credentialSubject
+			},
+			onError() {
+				toastsError({
+					msg: { text: i18Copy.auth.error.error_requesting_poh_credential }
+				});
+				reject();
+			},
+			onSuccess(response: VerifiablePresentationResponse) {
```

Comment:
is there a way to extract the `onSuccess` in a separate function for readability or it's a bit tricky because we need to pass resolve?

Just asking for reading the PR, maybe in the editor it's easier to follow

Reply:
Agree, done!